xsoar/resource_integration_instance.go

```diff
@@ -165,21 +211,45 @@ func (r resourceIntegrationInstance) Create(ctx context.Context, req tfsdk.Creat
 			//moduleInstance["outgoingMapperId"] = ""
 			//moduleInstance["passwordProtected"] = false
 			var propLabels []string
-			plan.PropagationLabels.ElementsAs(ctx, propLabels, false)
+			plan.PropagationLabels.ElementsAs(ctx, &propLabels, false)
 			moduleInstance["propagationLabels"] = propLabels
 			//moduleInstance["resetContext"] = false
 			moduleInstance["version"] = -1
 			break
 		}
 	}
 	var configs map[string]any
-	err = json.Unmarshal([]byte(plan.Config.Value), &configs)
-	if err != nil {
-		resp.Diagnostics.AddError(
-			"Error creating integration instance",
-			"Could not parse integration instance config json: "+err.Error(),
-		)
-		return
+	if plan.ConfigJson.Null {
+		configs = map[string]any{}
+	} else {
+		err = json.Unmarshal([]byte(plan.ConfigJson.Value), &configs)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error creating integration instance",
+				"Could not parse integration instance config json: "+err.Error(),
+			)
+			return
+		}
+	}
+	var secretConfigs map[string]any
+	var secretConfigJson string
+	if plan.SecretConfigJson.Null || plan.SecretConfigJson.Value == "" {
+		secretConfigJson = ""
+		secretConfigs = map[string]any{}
+	} else {
+		secretConfigJson = plan.SecretConfigJson.Value
+		err = json.Unmarshal([]byte(plan.SecretConfigJson.Value), &secretConfigs)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error creating integration instance",
+				"Could not parse integration instance secret config json: "+err.Error()+"\n\""+plan.SecretConfigJson.Value+"\"",
+			)
+			return
+		}
+	}
+	for key, element := range secretConfigs {
+		//TODO: add check for key already existing in configs
```

Comment:
What do we do to make sure this is actually done? I think it's nice to include some indication when this will be done, or what will resolve it (e.g. ticket number or just a comment saying "I'll check it in the remainder of this user story") .

Reply:
I implemented the check